### PR TITLE
User can see a list of vocabulary they've encountered

### PR DESCRIPTION
**Pull request by nadintamer**
## Description
This PR adds an initial interface for the following user stories:
* Tapping on a word adds it to vocabulary database
* User can see a list of vocabulary that they've encountered
* Each item in the vocabulary list displays the word in the target language, its English counterpart, as well as a flag representing the target language
* User can pull to refresh their vocabulary list

**Additional notes:**
* Vocabulary will display words from every target language that the user has previously selected
* Tapped words will only be added to the user's vocabulary if they don't already exist

## Testing
* Verified that changes don't break previous features like tab navigation, content feed, and viewing articles
* Verified that objects created through the app appear in the Parse database on Back4App

## Walkthrough GIF
vocabulary

### CODE REVIEW

File: app/src/main/java/com/example/lexis/fragments/PracticeFragment.java [hunk 1]
```diff
@@ -2,65 +2,82 @@
...
+        adapter = new VocabularyAdapter(this, vocabulary);
+        binding.rvVocabulary.setAdapter(adapter);
+        binding.rvVocabulary.setLayoutManager(new LinearLayoutManager(getActivity()));
+
+        // set up pull to refresh
+        binding.swipeContainer.setOnRefreshListener(() -> {
+            adapter.clear();
+            queryVocabulary();
+            binding.swipeContainer.setRefreshing(false);
+        });
```
Comment (annechen112995) [inline app/src/main/java/com/example/lexis/fragments/PracticeFragment.java:63]: Nice!

File: app/src/main/java/com/example/lexis/utilities/Utils.java [hunk 2]
```diff
@@ -74,4 +83,38 @@ public static Calendar getYesterday() {
         cal.add(Calendar.DATE, -1);
         return cal;
     }
+
+    /*
+    Return the current target language of the logged-in user.
+    */
+    public static String getCurrentTargetLanguage() {
+        ParseQuery<ParseUser> query = ParseUser.getQuery();
+        query.whereEqualTo("objectId", ParseUser.getCurrentUser().getObjectId());
```
Comment (annechen112995) [inline app/src/main/java/com/example/lexis/utilities/Utils.java:92]: Does this field also need to be null checked? `ParseUser.getCurrentUser().getObjectId()`

**Review by annechen112995 — APPROVED** [review]
Looks great! Pending comments

File: app/src/main/java/com/example/lexis/utilities/Utils.java (hunk 2 above)
Comment (nadintamer) [reply app/src/main/java/com/example/lexis/utilities/Utils.java]: Fixed — I actually also ended up spotting + fixing a bug that could cause the app to crash when getting the target language fails, thank you!  😊

**Pull request merged**